File: Recursion/decimalToHexadecimal.py
//わからなかったためもう一度復習

import math
def decimalToHexadecimal(decNumber):
  list = "0123456789ABCDEF"
  answer = ''
  binary = 0

  while decNumber >= 0:
    binary = decNumber % 16
    answer = list[binary] + answer
    decNumber = math.floor(decNumber / 16)

    if decNumber == 0: break

  return answer

File: ControlFlow/decimalToHexadecimal.java
class Solution{
    public static String decimalToHexadecimal(int decNumber){
      
    int currentBinary = 0;
    String hexadecimal = "0123456789ABCDEF";
    String str = "";

    while (decNumber >= 0) {
      currentBinary = decNumber % 16;
      str = hexadecimal.charAt(currentBinary) + str; 
      decNumber = decNumber /= 16;
      if (decNumber == 0)
        break;

    }
    return str;
  }
    
}

